core-common/src/main/java/org/zalando/nakadi/repository/db/StorageDbRepository.java

```diff
@@ -41,11 +43,22 @@ public StorageDbRepository(final JdbcTemplate template, final ObjectMapper mappe
         return storages;
     }
 
+    public Optional<Storage> getDefaultStorage() throws RepositoryProblemException {
+        final List<Storage> storages = jdbcTemplate.query(
+                "SELECT " + STORAGE_FIELDS + " FROM zn_data.storage WHERE st_default=? order by st_id limit 1",
```

Comment:
Why do we store the default storage in the DB if it is defined by configuration?

Reply:
@tor-vs-floki 
Here you are touching the comment from `StorageService` init method. Default storage id was stored in zookeeper (and after this PR - in database), the configuration properties are working only for first initialization of the service (when there are no values present in database and zookeeper).

Reply:
Could we store only storage on the database, having the default storage id only configurable from the environment, and then we could only query the database by storage ID to ensure it exists and get the required information from the DB?

Reply:
@tor-vs-floki 
We are able to create storage at runtime, use timeline to switch to it. Why setting storage as default should require redeployment. In other words - where is the benefit from what you are suggesting?

The second part - why I do not like to couple together environment vars and database configuration. Right now default storage is created if it was not created yet, and configuration is mystically taken somewhere from environment. It means, that we constantly provide configuration, that is:
- always not used
- most probably is outdated (totally incorrect values written there).

In these places (where configuration is stored in db), I would tend to remove configuration from environment at all, not to have 2 different places to maintain (and remove ambiguity between them).
The problem with it right now is that our tests and publishing to kpi event types are relying on existence of storage during application startup. If there would be more time, I would remove configuration from environment and would rely on external configuration only.